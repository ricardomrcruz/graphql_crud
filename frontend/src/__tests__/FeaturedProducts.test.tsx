import { describe, it, expect } from "@jest/globals";
import { render, screen } from "@testing-library/react";
import FeaturedProducts from "../components/FeaturedProducts";
import React from "react";
import Image from "next/image";
import Link from "next/link";
import {
  GetAllProductsQuery,
  useGetAllProductsQuery,
} from "@/graphql/generated/schema";

describe("FeaturedProduct card component", () => {
  it("renders correctly", () => {
    const view = render(
      <div className="pt-24">
        <div className="text-white text-3xl  w-full text-center justify-center  ">
          <h1 className="text-left my-10">Meilleures Ventes Fleur CBD</h1>
        </div>
        <div className="  items-center grid grid-cols-4">
          <div className="inter-var w-full bg-gray-50 relative dark:hover:shadow-2xl  dark:bg-black dark:border-white/[0.2] border-black/[0.1]  h-[25rem] rounded-sm p-2 border">
            <Link href={`/products/1`}>
              <div className="text-xl font-bold text-neutral-600 dark:text-white absolute z-10 p-2 bg-black rounded-br-lg">
                19.99€
              </div>
              <div className="relative h-[85%] mx-auto">
                <Image
                  src="https://img.com/i.png"
                  layout="fill"
                  objectFit="cover"
                  alt="Mandarine Fleur CBD"
                  className="rounded-sm"
                />
              </div>
              <p className="text-neutral-500 text-xl max-w-sm mt-2 dark:text-neutral-300">
                amnesia
              </p>
              <p className="text-neutral-500 text-md max-w-sm dark:text-neutral-300">
                10% CBD • fruity
              </p>
              <div className="flex-1 max-h-[87%] w-full overflow-hidden relative mt-4 rounded-sm"></div>
            </Link>
          </div>
        </div>
      </div>
    );

    expect(view.baseElement).toMatchSnapshot();
  });
});
