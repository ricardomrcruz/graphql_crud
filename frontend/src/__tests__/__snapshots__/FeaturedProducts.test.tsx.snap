// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FeaturedProduct card component renders correctly 1`] = `
<body>
  <div>
    <div
      class="pt-24"
    >
      <div
        class="text-white text-3xl  w-full text-center justify-center  "
      >
        <h1
          class="text-left my-10"
        >
          Meilleures Ventes Fleur CBD
        </h1>
      </div>
      <div
        class="  items-center grid grid-cols-4"
      >
        <div
          class="inter-var w-full bg-gray-50 relative dark:hover:shadow-2xl  dark:bg-black dark:border-white/[0.2] border-black/[0.1]  h-[25rem] rounded-sm p-2 border"
        >
          <a
            href="/products/1"
          >
            <div
              class="text-xl font-bold text-neutral-600 dark:text-white absolute z-10 p-2 bg-black rounded-br-lg"
            >
              19.99€
            </div>
            <div
              class="relative h-[85%] mx-auto"
            >
              <img
                alt="Mandarine Fleur CBD"
                class="rounded-sm"
                data-nimg="fill"
                decoding="async"
                loading="lazy"
                sizes="100vw"
                src="/_next/image?url=https%3A%2F%2Fimg.com%2Fi.png&w=3840&q=75"
                srcset="/_next/image?url=https%3A%2F%2Fimg.com%2Fi.png&w=640&q=75 640w, /_next/image?url=https%3A%2F%2Fimg.com%2Fi.png&w=750&q=75 750w, /_next/image?url=https%3A%2F%2Fimg.com%2Fi.png&w=828&q=75 828w, /_next/image?url=https%3A%2F%2Fimg.com%2Fi.png&w=1080&q=75 1080w, /_next/image?url=https%3A%2F%2Fimg.com%2Fi.png&w=1200&q=75 1200w, /_next/image?url=https%3A%2F%2Fimg.com%2Fi.png&w=1920&q=75 1920w, /_next/image?url=https%3A%2F%2Fimg.com%2Fi.png&w=2048&q=75 2048w, /_next/image?url=https%3A%2F%2Fimg.com%2Fi.png&w=3840&q=75 3840w"
                style="position: absolute; height: 100%; width: 100%; left: 0px; top: 0px; right: 0px; bottom: 0px; object-fit: cover; color: transparent;"
              />
            </div>
            <p
              class="text-neutral-500 text-xl max-w-sm mt-2 dark:text-neutral-300"
            >
              amnesia
            </p>
            <p
              class="text-neutral-500 text-md max-w-sm dark:text-neutral-300"
            >
              10% CBD • fruity
            </p>
            <div
              class="flex-1 max-h-[87%] w-full overflow-hidden relative mt-4 rounded-sm"
            />
          </a>
        </div>
      </div>
    </div>
  </div>
</body>
`;
